examples/fnirs/plot_classif_fnirs.py

```diff
@@ -0,0 +1,482 @@
+    def _block_diag(matrices):
+        """Construct a block diagonal matrix from a list of square matrices."""
+        shapes = [m.shape[0] for m in matrices]
+        total_size = sum(shapes)
+        result = np.zeros((total_size, total_size), dtype=matrices[0].dtype)
+        start = 0
+        for m in matrices:
+            end = start + m.shape[0]
+            result[start:end, start:end] = m
+            start = end
+        return result
+
+
+###############################################################################
+# Download example data and plot
+# ------------------------------
+
+X_url = "https://zenodo.org/records/13841869/files/X.npy"
+y_url = "https://zenodo.org/records/13841869/files/y.npy"
+
+data_path = Path("./data")
+data_path.mkdir(exist_ok=True)
+X_path, y_path = data_path / "X.npy", data_path / "y.npy"
+
+
+# Function to download the files if they don't already exist
+def download_file(url, file_path):
+    if not os.path.isfile(file_path):
+        print(f"Downloading {file_path} from {url}")
+        urllib.request.urlretrieve(url, file_path)
+        print(f"Downloaded {file_path}")
+
+
+# Download
+download_file(X_url, X_path)
+download_file(y_url, y_path)
+
+# Load the dataset
+X, y = np.load(X_path), np.load(y_path)
+
+print(
+    f"Data loaded: {X.shape[0]} trials, {X.shape[1]} channels, "
+    f"{X.shape[2]} time points"
+)
+
+# Get trials with the label "OP" for "Own Paradigm"
+MT_label = "OP"
+MT_trials_indices = np.where(y == MT_label)[0]
+
+# Average the data across the "OP" trials
+X_MT_erp = np.mean(X[MT_trials_indices, :, :], axis=0)
+
+# select example channel
+channel_index = 2
+
+# Plot the averaged signals
+plt.figure(figsize=(10, 5))
+plt.plot(X_MT_erp[channel_index, :], label="HbO", color="red")
+plt.plot(X_MT_erp[channel_index + 62, :], label="HbR", color="blue")
+plt.xlabel("Time samples")
+plt.ylabel("Signal Amplitude")
+plt.title(f"ERP for OP [Own Paradigm] trials in channel {channel_index}")
+plt.legend()
+plt.show()
+
+###############################################################################
+# Set up the pipeline
+# -------------------
+
+# Define the pipeline with HybridBlocks and SVC classifier
+pipeline_hybrid_blocks = Pipeline(
+    [
+        ("block_kernels", HybridBlocks(block_size=block_size)),
+        ("classifier", SVC(metric="riemann")),
+    ]
+)
+
+# Define the pipeline with BlockCovariances and SVC classifier
+pipeline_blockcovariances = Pipeline(
+    [
+        ("covariances", BlockCovariances(block_size=block_size)),
+        ('shrinkage', Shrinkage()),
+        ("classifier", SVC(metric="riemann")),
+    ]
+)
+
+###############################################################################
+# Define hyperparameter cross-validation
+# --------------------------------------
+
+# Define the hyperparameters for fitting
+pipeline_hybrid_blocks.set_params(
+    block_kernels__shrinkage=[0.01, 0],
+    block_kernels__metrics=['cov', 'rbf'],
+    classifier__C=0.1
```

Comment:
Hyper-parameters with only one value should be removed from the grid search.

Same remark for `pipeline_blockcovariances`.